distributed/worker.py

```diff
@@ -3228,12 +3237,19 @@ def apply_function_actor(
 
     result = function(*args, **kwargs)
 
+    notify_task_finished(plugins, result)
+
     with active_threads_lock:
         del active_threads[ident]
 
     return result
 
 
+def notify_task_finished(plugins, result):
+    for plugin in plugins.values():
+        if hasattr(plugin, "task_finished"):
+            plugin.task_finished(result)
```

Comment:
it might also be helpful to have a state here (i.e. "finished successfully" vs "exception")